Sources/Target/Linux/Process.cpp

```diff
@@ -215,17 +215,57 @@ ErrorCode Process::wait() {
 
     switch (_currentThread->_stopInfo.event) {
     case StopInfo::kEventNone:
-      // If the thread is stopped for no reason, it means the debugger (ds2)
-      // sent a SIGSTOP to it while it was already stopped for another reason,
-      // the SIGSTOP was queued, and now we're getting this notification. We
-      // have to ignore this and just let the thread continue.
-      // If `stepping` is true, it means the thread was actually being
-      // single-stepped before stopping, so instead of doing a `resume()`, we
-      // have to do a new `step()`.
-      if (stepping) {
+      // If _stopInfo.event is kEventNone it means the thread is either
+      // 1. stopped because another thread of the inferior hit a breakpoint
+      //    and ds2 sent a SIGSTOP to pause this thread while the breakpoint
+      //    is evaluated. In this case we just resume the thread.
+      // 2. stopped because this thread called clone(2) (or similar e.g.
+      //    pthread_create). In this case we have two subcases:
+      //    a. The thread was running and hit the clone/spawn and thus was
+      //       send a SIGTRAP and stopped. We just resume it and goto
```

Comment:
`s/was send a SIGTRAP and stopped/stopped with a SIGTRAP/` (you have a typo with send, and technically, nobody sends anything, the kernel just stops the thread and reports the status as if it were stopped with a SIGTRAP.